packages/pnpm/src/cmd/recursive/run.ts

```diff
@@ -84,7 +85,7 @@ export default async <T> (
     )))
   }
 
-  if (scriptName !== 'test' && !hasCommand) {
+  if (scriptName !== 'test' && !hasCommand && opts.allPackagesAreSelected) {
```

Comment:
Do we print some info message in this case? I think something should be printed with logger.info

Like `None of the selected packages has a "${scriptName}" script`

Reply:
What should the `prefix` be for the log? I see that `workspacePrefix` is used in the default export for `recursive.ts`:

https://github.com/pnpm/pnpm/blob/d2dc3f29f37141bdd746cb349f7ffedea4849165/packages/pnpm/src/cmd/recursive/index.ts#L80

But the normalization on line 76 isn't put back into `opts.workspacePrefix`, so it's still potentially `undefined` when passed into the run command:

https://github.com/pnpm/pnpm/blob/d2dc3f29f37141bdd746cb349f7ffedea4849165/packages/pnpm/src/cmd/recursive/index.ts#L76

If we want to use `opts.workspacePrefix`, I think I could do any of the following:

- store the normalized prefix on `opts`, so its definitely defined (my preference)
- redo the normalization in `run.ts`: `{ prefix: opts.workspacePrefix || process.cwd() }`
- fall back to `opts.prefix`: `{ prefix: opts.workspacePrefix || opts.prefix }`

Reply:
> store the normalized prefix on opts, so its definitely defined (my preference)

that sounds right